ecosystem/sep-0034.md

```diff
@@ -0,0 +1,84 @@
+## Preamble
+
+```
+SEP: To Be Assigned
+Title: Client Application Attribution
+Author: Jake Urban, Leigh McCulloch
+Track: Standard
+Status: Draft
+Created: 2020-08-17
+Discussion: https://github.com/stellar/stellar-protocol/issues/685
+```
+
+## Simple Summary
+This protocol outlines how a client can prove to a web server that the application requesting the relevant operation is owned by the client.
+
+## Motivation
+In general, web services may want to attribute incoming requests to client applications on a per-operation basis. Because requests headers like `User-Agent` can be spoofed, anchors require a cryptographically secure method of verifying a client's identity.
+
+For example, an anchor server may not charge fees for transactions created using a particular wallet. In this case, the operation is processing a deposit or withdrawal transaction, but the operation could be any action related to different use cases.
+
+## Abstract
+This following protocol involves 3 parties:
+
+- the client application: the party requesting an operation from the attribution server 
+- the client's signing server: the web server that claims ownership of the client application
+- the attribution server: the web server that verifies the signing server's claims
+
+At a high level, the signing server generates a [JSON Web Signature](https://www.rfc-editor.org/rfc/rfc7515.html) (JWS) that the client receives and includes in future requests to the attribution server. The attribution server verifies that the token included in the request was signed by the signing server, proving that the signing server takes ownership for the client requesting the operation from the verfication server.
+
+How the client application communicates with the signing server and receives the JWS is out of the scope of this document. The only requirement is for the signing server to return a JWS in the format descibed in the next section.
```

Comment:
Why doesn't the SEP define this? If each anchor has their own API for accepting this JWS, it's going to be hard for clients to implement this feature.

Reply:
The client communicates with its own server to obtain the JWS, not the anchor it is trying to verify its identity with. We don't need to define the protocol between the client application and the client's signing server as long as the JWS is in the correct format. 

We could describe what a flow between the client application and signing server could look like, but it is useless to define a standard and expect it to be used since there is no need for interoperability.